### ISSUE TITLE
[BUG] pench-agent-0.71 release tarball related issues

### ISSUE THREAD

**Issue by pravins** [distributed-system-analysis/pbench#3012]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

I am trying to package pbench-agent-0.71 for Fedora. Following issues with release tarball.
1. Released tarball https://github.com/distributed-system-analysis/pbench/releases/tag/v0.71.0. Size of this is 176M and has whole repo instead of just pbench-agent. 
   As per Fedora packaging guideline, we have to reference Source as it is in SPEC file and SRPM. https://docs.fedoraproject.org/en-US/packaging-guidelines/SourceURL/
   This will make resulting tarball very big.
2. Checking SRPM of pbench-agent-0.71 from https://download.copr.fedorainfracloud.org/results/ndokos/pbench/fedora-35-x86_64/04590468-pbench-agent/pbench-agent-0.71.0-3g85910732a.src.rpm  It has custom release of pbench-agent-0.71.0.tar.gz  (360K)  but its not uploaded anywhere, so no reference download link for it. 
3.  Custom release of pbench-agent-0.71.0.tar.gz  (360K)  does not has LICENSE file included in it.

### To Reproduce

Download Release tabrall or SRPM from ndokos copr repo.

### Actual Results

Official release tarball is 176M and has many not required files for pbench-agent. 
Custom build pbench-agent-0.71.tar.gz is not officially uploaded anywhere and also does not have LICENSE file into it.

### Expected Results

Release pbench-agent-0.71 specific tarball and upload it at github

### Additional information

None

**Comment by webbnh** [comment]
Pravin, I don't think that there is any requirement or expectation that the "release" of the Pbench Agent from GitHub match (file for file) the contents of the SRPM used to build the Agent binary RPM.  The requirement for providing an SRPM which matches our RPM can and should be met using a different release mechanism, based on COPR (or whatever we use to build the RPM) or what we actually submit to it.

The GitHub release really has nothing to do with building RPMs or packaging for Fedora.

I concur, thought, that the SRPM should have a `LICENSE` file in it.

That said, making sure that we have a link to our SRPM is pretty much the very last step in getting ready for Fedora, so I think that there are other issues which we need to address first.

**Comment by pravins** [comment]
This requirement giving in Fedora packaging guidelines : https://docs.fedoraproject.org/en-US/packaging-guidelines/SourceURL/

In Fedora, we do not build from SRPM (Though its possible, we need to add some extra steps in SPEC file but no upstream releases SRPM, upsteam releases tarball and all distribution used tarball to build .dep or .rpm ), rather we build SRPM with SPEC and Tarball :)

* We already have a tarball within SRPM, i think we need to upload script for making pbench-agent-0.71.tar.gz in our git. 
* Upload tarball at github release or some other hosting place. (I think github might be already allowing attaching extra tarball with releases)

**Comment by webbnh** [comment]
> This requirement in given in Fedora packaging guidelines : https://docs.fedoraproject.org/en-US/packaging-guidelines/SourceURL/

I'm not sure exactly which part of that page you're referring to.

> In Fedora, we do not build from SRPM (Though its possible, we need to add some extra steps in SPEC file but no upstream releases SRPM, upsteam releases tarball and all distribution used tarball to build .dep or .rpm ), rather we build SRPM with SPEC and Tarball :)

In Pbench, we build an SRPM locally and then hand that to COPR to have it made into RPM.  When we do the RPM builds without COPR (e.g., in the new CI support), we build them from a local set of sources, but those sources are a carefully-constructed temporary directory tree -- they are not the Git checkout; and, while the process does use `tar`, I don't think what it produces qualifies as a "source tarball" in RPM-building terms.

> We already have a tarball within SRPM, i think we need to upload script for making pbench-agent-0.71.tar.gz in our git.

I'm not sure what you're suggesting, here.  (Yes, I believe that an SRPM contains a "source tarball", so, yes, if we distributed an SRPM, the source tarball could be extracted from it; yes, we could provide a script for that, but I expect that that would be reinventing the wheel.)

> Upload tarball at github release or some other hosting place. (I think github might be already allowing attaching extra tarball with releases)

My understanding of open-source practice is that we are supposed to make available our sources along with our binaries.  I'm not sure of the details of what we have to do to comply with that, but I don't think it will be a problem.

However, as I've said, publishing our sources really isn't important until after we've managed all of the other changes required to be packaged for Fedora, so I really feel like pouring a lot of attention into this now is premature.

**Comment by webbnh** [comment]
> This requirement in given in Fedora packaging guidelines : https://docs.fedoraproject.org/en-US/packaging-guidelines/SourceURL/

I'm not sure exactly which part of that page you're referring to.

> In Fedora, we do not build from SRPM (Though its possible, we need to add some extra steps in SPEC file but no upstream releases SRPM, upsteam releases tarball and all distribution used tarball to build .dep or .rpm ), rather we build SRPM with SPEC and Tarball :)

In Pbench, we build an SRPM locally and then hand that to COPR to have it made into RPM.  When we do the RPM builds without COPR (e.g., in the new CI support), we build them from a local set of sources, but those sources are a carefully-constructed temporary directory tree -- they are not the Git checkout; and, while the process does use `tar`, I don't think what it produces qualifies as a "source tarball" in RPM-building terms.

> We already have a tarball within SRPM, i think we need to upload script for making pbench-agent-0.71.tar.gz in our git.

I'm not sure what you're suggesting, here.  (Yes, I believe that an SRPM contains a "source tarball", so, yes, if we distributed an SRPM, the source tarball could be extracted from it; yes, we could provide a script for that, but I expect that that would be reinventing the wheel.)

> Upload tarball at github release or some other hosting place. (I think github might be already allowing attaching extra tarball with releases)

My understanding of open-source practice is that we are supposed to make available our sources along with our binaries.  I'm not sure of the details of what we have to do to comply with that, but I don't think it will be a problem.

However, as I've said, publishing our sources really isn't important until after we've managed all of the other changes required to be packaged for Fedora, so I really feel like pouring a lot of attention into this now is premature.

**Comment by pravins** [comment]
You are right Webb, not much benefits into looking into it at this moment. 

For trail and error, i have copied tarball from Peter's copr repo into https://pravins.fedorapeople.org/pbench-agent-0.72.0.tar.gz 
Once we have something concrete, i will let you know :)

**Comment by webbnh** [comment]
If you are interested, I think I can show you how to build your own "source tarball" from your favorite Pbench Git checkout.

**Comment by pravins** [comment]
Thats great. Lets do that
Thanks :)
